staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -312,6 +312,11 @@ func (f *Cloud) InstanceExistsByProviderID(ctx context.Context, providerID strin
 // InstanceShutdownByProviderID returns true if the instances is in safe state to detach volumes
 func (f *Cloud) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
 	f.addCall("instance-shutdown-by-provider-id")
+
+	if providerID == "" {
+		return false, fmt.Errorf("cannot shutdown instance with empty providerID")
+	}
```

Comment:
This is not strictly needed here, but it's to show that the previous version of the code would have failed the unit test I'm adding right now. I could drop it because it's no-op now, but for the sake of avoiding regressions in the future it may be worth keeping

Reply:
if that mimics that mimics the cloud-providers implementations it seems ok, is this a general behavior?

Reply:
Today when we call `shutdownInCloudProvider` with empty providerID it blindly calls `InstanceShutdownByProviderID` passing the empty ID. The former exists only in Azure, GCE and vSphere provider.

In Azure for empty providerID we return `false, nil` (so no error thrown).

In GCP the function is not implemented and always returns `false, cloudprovider.NotImplemented` (so technically an error)

In vSphere the function immediately calls `GetNodeNameFromProviderID(providerID)` and that one for empty argument will return an error `Error while getting nodename for providerID none` (which is somehow misleading).

So overall the behaviour is inconsistent but in majority of cases (2 out of 3) we error if try to shutdown a nil-instance.